### DISCUSSION TITLE
LAB 06.02 Task 3

### DISCUSSION THREAD

**Discussion by joanfco30** [rramosp/ai4eng.v1#141] [06 - MÉTODOS NO SUPERVISADOS]
Buenos noches un cordial saludo. Lo primero disculpas para las dudas reiterativas y especialmente en este laboratorio. 

Estaba haciendo la parte tres de este laboratorio y note algo al momento del grade. 
plot_faces(faces[find_similar(face, faces, 2)]) cuando cambie valor de n = 30 a n = 2 el grade fue cinco, la verdad creo que fue coincidencia. 
Tengo una inquietud con esta última tarea ya que no estoy entiendo que error tengo al momento de hacer la evaluación.
`def find_similar(v,X,n):
    from sklearn.decomposition import NMF
    
    nmf = NMF(n_components=30, init="random", random_state=0)
    nmf = nmf.fit(X) ## your code here. call the 'fit' method
    Xt = nmf.fit_transform(X) # use nmf to transform X
    vt = nmf.fit_transform(v.reshape(1,-1)) # use nmf to transform v .. you will have to use reshape like this v.reshape(1,-1)
    result = [np.linalg.norm(Xt[i]-vt) for i in range(len(Xt))]
    print(len(X))
    result = np.argsort(result)[0:n:1]
    print(result)
    #result = ... # your code here
    
    return result`

Pero mi resultado es el siguiente. 
![image](https://user-images.githubusercontent.com/60493083/173973418-00e36c7a-0066-4f01-9308-edc387d84294.png)

Seguramente hay algo que este haciendo mal en este procedimiento o que me falta en la función que estoy implementando. 

Muchas gracias y quedo atento a sus sugerencias.

**Comment by joanfco30** [comment]
Ya solucioné, el error era demasiado evidente. Nuevamente mil gracias.

`Xt = nmf.fit_transform(X) # use nmf to transform X

vt = nmf.fit_transform(v.reshape(1,-1)) # use nmf to transform v .. you will have to use reshape like this v.reshape(1,-1)`

**Reply by Felipe-RA** [reply]
Gracias por compartir tu solución!
